cmd/swarm/main.go

```diff
@@ -378,7 +384,7 @@ func registerBzzService(ctx *cli.Context, stack *node.Node) {
 		} else {
 			swapEnabled = false
 		}
-		return swarm.NewSwarm(ctx, client, bzzconfig, swapEnabled, syncEnabled, cors, pssEnabled)
+		return swarm.NewSwarm(ctx, client, bzzconfig, swapEnabled, syncEnabled, cors, pssEnabled, discoveryEnabled)
```

Comment:
`discoveryEnabled` should be used to modify the config, rather than being passed to `NewSwarm`.